File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        l, r = 0, len(s) - 1
        while l < r and s[l] == s[r]:
            while l + 1 < r and s[l + 1] == s[l]:
                l += 1
            while r - 1 > l and s[r - 1] == s[r]:
                r -= 1
            l += 1
            r -= 1
        return r - l + 1
        

File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumLength = function(s) {
  let [l, r] = [0, s.length - 1];
  
  while (l < r && s[l] === s[r]) {
    while (l + 1 < r && s[l + 1] === s[l]) l++;
    while (r - 1 > l && s[r - 1] === s[r]) r--;
    
    l++;
    r--;
  }
  
  return r - l + 1;
};